File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dictionary = {}
        for word in strs:
            print(''.join(sorted(word)))
            sortedWord = ''.join(sorted(word))
            # If word is not in dictionary
            if sortedWord not in dictionary:
                dictionary[sortedWord] = [word]
            # If previously it is present that means its anagram
            # was previously present
            else:
                dictionary[sortedWord] += [word]
        return [dictionary[i] for i in dictionary]

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
 let dictionary = {}
    for (s of strs){
//         for every word in arr, reorder the char in words
        key = reorderStr(s);
        if (dictionary[key] === undefined){
            dictionary[key] = [s]
        } else {
            dictionary[key] = dictionary[key].concat(s)
        }
    }
    return Object.values(dictionary)
};

function reorderStr(str) {
	return str.split('').sort().join('');
}